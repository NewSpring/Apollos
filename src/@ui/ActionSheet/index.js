export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
