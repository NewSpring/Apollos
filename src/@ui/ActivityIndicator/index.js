import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
