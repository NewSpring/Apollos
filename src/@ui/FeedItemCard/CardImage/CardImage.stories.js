import React, { Component } from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';

import CardImage from './';

storiesOf('@ui/FeedItemCard/CardImage', module)
  .add('Default', () => {
    const centered = {
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1,
    };

    return (
      <View style={centered}>
        <CardImage source={'https://picsum.photos/600/400/?random'} />
      </View>
    );
  })
  .add('With Overlay', () => {
    const centered = {
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1,
    };

    return (
      <View style={centered}>
        <CardImage source={'https://picsum.photos/600/400/?random'} overlayColor={'salmon'} />
      </View>
    );
  })
  .add('isLoading', () => React.createElement(class CardImageLoading extends Component {
    constructor() {
      super();
      this.state = { isLoading: true };
    }

    componentDidMount() {
      setTimeout(() => {
        this.setState({ isLoading: false });
      }, 4000);
    }

    render() {
      const centered = {
        justifyContent: 'center',
        alignItems: 'center',
        flex: 1,
      };

      return (
        <View style={centered}>
          <CardImage isLoading={this.state.isLoading} source={'https://picsum.photos/600/400/?random'} />
        </View>
      );
    }
  }));
