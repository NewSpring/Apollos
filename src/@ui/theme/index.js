export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
