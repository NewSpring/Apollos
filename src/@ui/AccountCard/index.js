import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
