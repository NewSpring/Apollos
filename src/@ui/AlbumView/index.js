import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
