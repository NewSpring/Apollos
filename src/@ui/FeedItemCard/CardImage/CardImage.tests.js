import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '@ui/theme';
import CardImage from './';

describe('the FeedItemCard Category component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <CardImage source={'https://picsum.photos/600/400/?random'} overlayColor={'#ffffff'} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
