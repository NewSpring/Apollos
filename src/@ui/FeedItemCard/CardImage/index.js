import React from 'react';
import { StyleSheet, View, Platform } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import Color from 'color';

import { withTheme } from '@ui/theme';
import styled from '@ui/styled';
import { CardImage } from '@ui/Card';
import LinearGradient from '@ui/LinearGradient';

const enhance = compose(
  pure,
  withTheme(),
  setPropTypes({
    source: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.shape({
        uri: PropTypes.string,
        label: PropTypes.string,
        width: PropTypes.number,
        height: PropTypes.number,
      })),
      PropTypes.string,
    ]),
    overlayColor: PropTypes.string,
  }),
);

const Wrapper = styled(({ theme }) => ({
  ...Platform.select({
    android: { // fixes android borderRadius overflow display issue
      borderTopRightRadius: theme.sizing.borderRadius,
      borderTopLeftRadius: theme.sizing.borderRadius,
    },
  }),
}))(View);

const Overlay = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  ...Platform.select({
    android: { // fixes android borderRadius overflow display issue
      borderTopRightRadius: theme.sizing.borderRadius,
      borderTopLeftRadius: theme.sizing.borderRadius,
    },
  }),
}))(LinearGradient);

const FeedItemCardImage = enhance(({
  source: imageSource,
  overlayColor,
}) => (
  <Wrapper>
    <CardImage source={imageSource} />
    {overlayColor ? <Overlay colors={[`${Color(overlayColor).fade(1).string()}`, overlayColor]} start={[0, 0]} end={[0, 1]} locations={[0.3, 1]} /> : null}
  </Wrapper>
));

export default FeedItemCardImage;
