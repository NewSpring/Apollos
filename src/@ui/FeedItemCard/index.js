import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { startCase, toLower } from 'lodash';

import { withThemeMixin, withTheme } from '@ui/theme';
import Icon from '@ui/Icon';
import CategoryLabel from '@ui/CategoryLabel';
import Card, { CardContent, CardActions } from '@ui/Card';
import { H4 } from '@ui/typography';
import styled from '@ui/styled';

import CardImage from './CardImage';
import LikeButton from './LikeButton';

const StyledCardContent = styled({ paddingBottom: 0 })(CardContent);

const enhance = compose(
  pure,
  defaultProps({
    isLight: true,
  }),
  withThemeMixin(({ isLight }) => ({
    type: isLight ? 'light' : 'dark',
  })),
  withTheme(),
  setPropTypes({
    title: PropTypes.string.isRequired,
    images: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        value: PropTypes.string,
        description: PropTypes.string,
      })),
      PropTypes.string,
    ]),
    category: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    isLiked: PropTypes.bool,
    isLight: PropTypes.bool,
    backgroundColor: PropTypes.string,
    style: PropTypes.any, // eslint-disable-line
  }),
);

const FeedItemCard = enhance(({
  images,
  title,
  category,
  isLoading,
  isLiked,
  backgroundColor,
  theme,
  id,
  ...otherProps
}) => (
  <Card isLoading={isLoading} cardColor={backgroundColor} {...otherProps}>
    <CardImage source={images} overlayColor={backgroundColor} />
    <StyledCardContent>
      <H4>{startCase(toLower(title))}</H4>
    </StyledCardContent>
    <CardActions>
      <CategoryLabel
        label={startCase(toLower(category))}
        isLoading={isLoading}
        withFlex
      />
      <LikeButton id={id}>
        <Icon
          name={isLiked ? 'like-solid' : 'like'}
          size={theme.helpers.rem(1.2)}
          fill={theme.colors.text.primary}
          isLoading={isLoading}
        />
      </LikeButton>
    </CardActions>
  </Card>
));

export default FeedItemCard;
