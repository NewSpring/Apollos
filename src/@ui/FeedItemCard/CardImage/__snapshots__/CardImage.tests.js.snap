// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FeedItemCard Category component should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    maintainAspectRatio={false}
    source={
      Array [
        Object {
          "uri": "https://picsum.photos/600/400/?random",
          "url": "https://picsum.photos/600/400/?random",
        },
      ]
    }
    style={
      Array [
        Object {},
        Object {
          "aspectRatio": 1,
          "flex": 1,
          "resizeMode": "cover",
          "width": "100%",
        },
      ]
    }
  />
  <ExponentLinearGradient
    colors={
      Array [
        16777215,
        4294967295,
      ]
    }
    end={
      Array [
        0,
        1,
      ]
    }
    locations={
      Array [
        0.3,
        1,
      ]
    }
    start={
      Array [
        0,
        0,
      ]
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
